src/environment.rs

```diff
@@ -117,3 +134,47 @@ impl EnvironmentBuilder {
 pub struct EnvironmentOptions {
     pub(crate) client_options: ClientOptions,
 }
+
+/** Helper for tls configuration */
+#[derive(Clone, Copy)]
+pub struct TlsConfiguration {
+    pub(crate) enabled: bool,
+    pub(crate) hostname_verification: bool,
+    pub(crate) trust_everything: bool,
+}
```

Comment:
For the `TlsConfiguration` probably it's better if we do a builder pattern?

Something like this

```rust
pub struct TlsConfigurationBuilder(TlsConfiguration);

impl TlsConfigurationBuilder {

   pub fn trust_everything(mut self, trust_everything: bool)  -> TlsConfigurationBuilder {
        self.0.trust_everything = trust_everything;
        self
    }

    pub fn enable(mut self, enable: bool)  -> TlsConfigurationBuilder {
        self.0.enabled = enabled;
        self
    }


    pub fn hostname_verification_enable(mut self, hostname_verification: bool) -> TlsConfigurationBuilder  {
        self.0.hostname_verification = hostname_verification;
        self
    }

    pub fn build(self) -> TlsConfiguration {
       self.0
    }
}


impl TlsConfiguration {

   pub fn builder() -> TlsConfigurationBulder {
    TlsConfigurationBulder(TlsConfiguration::default())
   }
  
}
```